lib/matplotlib/figure.py

```diff
@@ -325,6 +325,10 @@ def __init__(self,
             frameon = rcParams['figure.frameon']
 
         self.bbox_inches = Bbox.from_bounds(0, 0, *figsize)
+        if not all(np.isfinite(_) for _ in figsize):
```

Comment:
Why not check this before the expense of creating a `Bbox` (not that I know if that's a very large expense or not)?